compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1434,13 +1709,53 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             debug!(?collected_lifetimes);
             debug!(?new_remapping);
 
+            lctx.create_generic_defs(opaque_ty_def_id, &ast_generics.params, &mut new_remapping);
+            debug!(?new_remapping);
+
+            debug!("impl_trait_inputs = {:#?}", impl_trait_inputs);
+            for ty in impl_trait_inputs {
+                if let TyKind::ImplTrait(node_id, _) = &ty.kind {
+                    let old_def_id = lctx.local_def_id(*node_id);
+
+                    let node_id = lctx.next_node_id();
+                    // Add a definition for the generic param def.
+                    let new_def_id =
+                        lctx.create_def(opaque_ty_def_id, node_id, DefPathData::ImplTrait);
+
+                    new_remapping.insert(old_def_id, new_def_id);
+                } else {
+                    unreachable!(
+                        "impl_trait_inputs contains {:?} which is not TyKind::ImplTrait(..)",
+                        ty.kind
+                    );
+                }
+            }
+
+            let in_scope_generics: Vec<_> = ast_generics
```

Comment:
Why do we need this?

Reply:
This was answered in https://github.com/rust-lang/rust/pull/101345#discussion_r963098364 or are you referring to why is it constructed here?.